currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/JoinDAOFragment.kt

```diff
@@ -170,6 +166,12 @@ class JoinDAOFragment() : BaseFragment(R.layout.fragment_join_network) {
         disableRefresher()
     }
 
+    fun parseTransactionDataGetWalletId(trans:TrustChainTransaction) : String {
+        val transaction = trans["message"].toString()
```

Comment:
Is it me or seems it unlogical what you are doing here? parse object to string and then back to an object, can't you do something like `trans["message"]["transaction"]["SW_UNIQUE_ID"] `?